test/vintage_net_bridge_test.exs

```diff
@@ -57,37 +57,72 @@ defmodule VintageNetBridgeTest do
+        %RawConfig{
+          ifname: "br0",
+          type: VintageNetBridge,
+          source_config: input,
+          required_ifnames: [],
+          child_specs: [
+            {VintageNetBridge.Server,
+             %{brctl: "brctl", bridge_ifname: "br0", interfaces: ["eth0", "mesh0"]}},
+            Utils.udhcpc_child_spec("br0", "unit_test"),
+            {VintageNet.Connectivity.InternetChecker, "br0"}
+          ],
+          down_cmds: [
+            {:run, "brctl", ["delbr", "br0"]},
+            {:run_ignore_errors, "ip", ["addr", "flush", "dev", "br0", "label", "br0"]},
+            {:run, "ip", ["link", "set", "br0", "down"]}
+          ],
+          up_cmds: [
+            {:run, "brctl", ["addbr", "br0"]},
+            {:run, "brctl", ["setbridgeprio", "br0", "2"]},
+            {:run, "brctl", ["setfd", "br0", "1"]},
+            {:run, "brctl", ["sethello", "br0", "3"]},
+            {:run, "brctl", ["setmaxage", "br0", "4"]},
+            {:run, "brctl", ["setpathcost", "br0", "5"]},
+            {:run, "brctl", ["setportprio", "br0", "6"]},
+            {:run, "brctl", ["hairpin", "br0", "7", "no"]},
+            {:run, "brctl", ["stp", "br0", "yes"]},
+            {:run_ignore_errors, "brctl", ["addif", "br0", "eth0"]},
+            {:run_ignore_errors, "brctl", ["addif", "br0", "mesh0"]},
+            {:run, "ip", ["link", "set", "br0", "up"]}
+          ]
+        }
+      else
+        %RawConfig{
+          ifname: "br0",
+          type: VintageNetBridge,
+          source_config: input,
+          required_ifnames: [],
+          child_specs: [
+            {VintageNetBridge.Server,
+             %{brctl: "brctl", bridge_ifname: "br0", interfaces: ["eth0", "mesh0"]}},
+            Utils.udhcpc_child_spec("br0", "unit_test"),
+            {VintageNet.Connectivity.InternetChecker, "br0"}
+          ],
+          down_cmds: [
+            {:run, "brctl", ["delbr", "br0"]},
+            {:run_ignore_errors, "ip", ["addr", "flush", "dev", "br0", "label", "br0"]},
+            {:run, "ip", ["link", "set", "br0", "down"]}
+          ],
+          up_cmds: [
+            {:run, "brctl", ["addbr", "br0"]},
+            {:run, "brctl", ["setfd", "br0", "1"]},
+            {:run, "brctl", ["hairpin", "br0", "7", "no"]},
+            {:run, "brctl", ["sethello", "br0", "3"]},
+            {:run, "brctl", ["setmaxage", "br0", "4"]},
+            {:run, "brctl", ["setpathcost", "br0", "5"]},
+            {:run, "brctl", ["setportprio", "br0", "6"]},
+            {:run, "brctl", ["setbridgeprio", "br0", "2"]},
+            {:run, "brctl", ["stp", "br0", "yes"]},
+            {:run_ignore_errors, "brctl", ["addif", "br0", "eth0"]},
+            {:run_ignore_errors, "brctl", ["addif", "br0", "mesh0"]},
+            {:run, "ip", ["link", "set", "br0", "up"]}
+          ]
+        }
+      end
 
     assert output == VintageNetBridge.to_raw_config("br0", input, Utils.default_opts())
```

Comment:
One way that you could split this up instead without the conditional would be to pull out the nondeterministic part and iterate it separately:

```elixir
assert %RawConfig{
          ifname: "br0",
          type: VintageNetBridge,
          source_config: input,
          required_ifnames: [],
          child_specs: [
            {VintageNetBridge.Server,
             %{brctl: "brctl", bridge_ifname: "br0", interfaces: ["eth0", "mesh0"]}},
            Utils.udhcpc_child_spec("br0", "unit_test"),
            {VintageNet.Connectivity.InternetChecker, "br0"}
          ],
          down_cmds: [
            {:run, "brctl", ["delbr", "br0"]},
            {:run_ignore_errors, "ip", ["addr", "flush", "dev", "br0", "label", "br0"]},
            {:run, "ip", ["link", "set", "br0", "down"]}
          ],
          up_cmds: up_cmds
        } == VintageNetBridge.to_raw_config("br0", input, Utils.default_opts())

  expected_up_cmds = [
            {:run, "brctl", ["addbr", "br0"]},
            {:run, "brctl", ["setfd", "br0", "1"]},
            {:run, "brctl", ["hairpin", "br0", "7", "no"]},
            {:run, "brctl", ["sethello", "br0", "3"]},
            {:run, "brctl", ["setmaxage", "br0", "4"]},
            {:run, "brctl", ["setpathcost", "br0", "5"]},
            {:run, "brctl", ["setportprio", "br0", "6"]},
            {:run, "brctl", ["setbridgeprio", "br0", "2"]},
            {:run, "brctl", ["stp", "br0", "yes"]},
            {:run_ignore_errors, "brctl", ["addif", "br0", "eth0"]},
            {:run_ignore_errors, "brctl", ["addif", "br0", "mesh0"]},
            {:run, "ip", ["link", "set", "br0", "up"]}
          ]
  assert length(up_cmds) == length(expected_up_cmds)
  assert Enum.all?(expected_up_cmds, & &1 in up_cmds)
```

Reply:
This is a great idea! I'll adjust the test.